lib/sync/index.js

```diff
@@ -98,43 +99,53 @@ function invoke(dataset_id, params, callback) {
   });
 }
 
-// extend from empty object to force copy, and because last source overrides the previous ones
-module.exports = _.extend({}, server, {
-  sync: metricsModule.timeAsyncFunc(metricsModule.KEYS.SYNC_API_PROCESS_TIME, server.sync),
-  syncRecords: metricsModule.timeAsyncFunc(metricsModule.KEYS.SYNC_API_PROCESS_TIME, server.syncRecords),
-  listCollisions: metricsModule.timeAsyncFunc(metricsModule.KEYS.SYNC_API_PROCESS_TIME, server.api.listCollisions),
-  removeCollision: metricsModule.timeAsyncFunc(metricsModule.KEYS.SYNC_API_PROCESS_TIME, server.api.removeCollision),
-  setLogLevel: setLogLevel,
-  api: _.extend({}, server.api, {
-    connect: connect,
-    invoke: invoke,
-    toJSON: toJSON,
+module.exports = function(cfg) {
```

Comment:
+1. @david-martin I think you can incorporate this change into PR #101. This way we can access the event emitter in here and allow the sync framework to emit more events.